extensions/Rules/Extension.pm

```diff
@@ -0,0 +1,120 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# This Source Code Form is "Incompatible With Secondary Licenses", as
+# defined by the Mozilla Public License, v. 2.0.
+
+package Bugzilla::Extension::Rules;
+
+use 5.10.1;
+use strict;
+use warnings;
+use parent qw(Bugzilla::Extension);
+
+use Try::Tiny;
+use TOML qw(from_toml);
+
+use Bugzilla::Constants;
+use Bugzilla::Logging;
+use Bugzilla::Status;
+use Bugzilla::Util qw(datetime_from);
+
+use Bugzilla::Extension::Rules::Rule;
+
+our $VERSION = '0.01';
+
+sub config_add_panels {
+  my ($self, $args) = @_;
+  my $modules = $args->{panel_modules};
+  return $modules->{Rules} = 'Bugzilla::Extension::Rules::Config';
+}
+
+sub bug_check_can_change_field {
+  my ($self, $args) = @_;
+  my ($bug, $field, $new_value, $old_value, $priv_results)
+    = @$args{qw(bug field new_value old_value priv_results)};
+  my $user          = Bugzilla->user;
+  my $request_cache = Bugzilla->request_cache;
+
+  my $rules_enabled = Bugzilla->params->{change_field_rules_enabled};
+  my $rules_toml    = Bugzilla->params->{change_field_rules};
+
+  if ($rules_enabled && $rules_toml) {
+    my $rule_defs;
+    try {
+      $rule_defs = $request_cache->{rule_defs} ||= from_toml($rules_toml);
```

Comment:
To avoid re-parsing the rules toml on every update request, it might be better to store the parsed and validated rules in memcache, similar to what we do for params.